### ISSUE TITLE
"unrecognized `--disable-executable-profiling'" with sandbox and -w flag

### ISSUE THREAD

**Issue by jberryman** [haskell/cabal#2313]
I'm trying to build a package for GHC 7.10 RC1 and getting errors I don't understand. To reproduce:

    jberryman /tmp » cabal --no-require-sandbox  update                                                                                                                                                                                      1 ↵
    Downloading the latest package list from hackage.haskell.org
    jberryman /tmp » cabal get atomic-primops
    Downloading atomic-primops-0.6.1.1...
    Unpacking to atomic-primops-0.6.1.1/
    jberryman /tmp » cd atomic-primops-0.6.1.1 
    jberryman /tmp/atomic-primops-0.6.1.1 » cabal sandbox init                                                                                                                                                                               1 ↵
    Writing a default package environment file to
    /tmp/atomic-primops-0.6.1.1/cabal.sandbox.config
    Creating a new sandbox at /tmp/atomic-primops-0.6.1.1/.cabal-sandbox
    jberryman /tmp/atomic-primops-0.6.1.1 » cabal install -w /usr/local/bin/ghc-7.10.0.20141222 --only-dependencies
    Resolving dependencies...
    Notice: installing into a sandbox located at
    /tmp/atomic-primops-0.6.1.1/.cabal-sandbox
    Configuring primitive-0.5.4.0...
    Failed to install primitive-0.5.4.0
    Build log ( /tmp/atomic-primops-0.6.1.1/.cabal-sandbox/logs/primitive-0.5.4.0.log ):
    unrecognized 'configure' option `--disable-executable-profiling'
    cabal: Error: some packages failed to install:
    primitive-0.5.4.0 failed during the configure step. The exception was:
    ExitFailure 1

I believe I was able to use `-w` before with sandboxes painlessly in the past, but I may be doing something wrong here. Here is my version info:

     jberryman /tmp/atomic-primops-0.6.1.1 » cabal --version
    cabal-install version 1.20.0.6
    using version 1.20.0.3 of the Cabal library

**Comment by crockeea** [comment]
I'm getting a similar error when installing Cabal-1.22.0.0:

```
...
[78 of 78] Compiling Main             ( /tmp/Cabal-1.22.0.0-9051/Cabal-1.22.0.0/dist/setup/setup.hs, /tmp/Cabal-1.22.0.0-9051/Cabal-1.22.0.0/dist/setup/Main.o )
Linking /tmp/Cabal-1.22.0.0-9051/Cabal-1.22.0.0/dist/setup/setup ...
unrecognized 'configure' option `--enable-executable-profiling'
cabal: Error: some packages failed to install:
Cabal-1.22.0.0 failed during the configure step. The exception was:
ExitFailure 1
```

**Comment by ttuegel** [comment]
`--enable-executable-profiling` is now just `--enable-profiling`. It was dishonestly named in the first place, because you need library profiling enabled, too. You probably have `executable-profiling: True` in your `$HOME/.cabal/config`, which now should be `profiling: True`.
https://github.com/haskell/cabal/blob/master/Cabal/changelog

**Comment by tibbe** [comment]
We should keep the old flash as well for backwards compat.

**Comment by 23Skidoo** [comment]
I thought we did.

**Comment by ttuegel** [comment]
I thought we kept, too. Fixing that now...

**Comment by ttuegel** [comment]
OK, I fixed this in master, and I backported it to the 1.22 branch, since this is a regression from 1.20.

**Comment by tibbe** [comment]
Thanks!

**Comment by hesselink** [comment]
I've just compiled the 1.22 branch, and while this is fixed for `--enable-executable-profiling`, it isn't for `--enable-library-coverage`:

```
$ cabal configure --enable-library-coverage
Resolving dependencies...
cabal: Option --enable-library-coverage is obsolete! Please use
--enable-coverage instead.
```

This seems to be coming from [here](https://github.com/haskell/cabal/blob/f2a07d2cd181540e7cc6c49b7f88178fe1eb4c6e/Cabal/Distribution/Simple/Configure.hs#L302).

**Comment by tibbe** [comment]
@ttuegel can you fix coverage as well please?

**Comment by tibbe** [comment]
This is a bit trickier than I initially thought. @dcoutts told me that cabal-install needs to check which Cabal version it ended up compiling the setup executable against and then make sure it passes flags that this version understands.

@dcoutts is there an example of us already doing this?

**Comment by ttuegel** [comment]
I think now it was a mistake to change the semantics of the `ConfigFlags` field behind these options. I am going to bring back the old fields with the old semantics and *add* new fields with the correct semantics (for the new, short-name options). It's not enough to make `--enable-executable-profiling` etc. synonyms for the new options because old scripts could expect the old semantics, wrong as they are. This will also let us warn the user if they invoke the old option.

As for cabal-install, besides passing each Cabal version only the flags it understands, we should also make cabal-install-1.22 replicate the new semantics for `--enable-profiling` etc. with the old flags.

**Comment by tibbe** [comment]
Fixed by 66dade371e720efd57b06424cfdf68454e6aba77 and f445c3a3bb02453ba6545cc4e9221073dfc9635d.
